packages/ast-spec/tests/util/serializers/string.ts

```diff
@@ -0,0 +1,38 @@
+import { NewPlugin } from 'pretty-format';
+
+// custom string serializer so that we can use single-quoted strings instead of double quoted strings
+// this plays nicer with the way that the snapshot diff result, which is a pure string
+const serializer: NewPlugin = {
+  test(val: unknown) {
+    return typeof val === 'string';
+  },
+  serialize(
+    str: string,
+    // config,
+    // indentation,
+    // depth,
+    // refs,
+    // printer,
+  ) {
+    const characters: string[] = [];
+
+    characters.push("'");
+    for (const character of str) {
+      switch (character) {
+        case "'":
+          characters.push('\\');
+          break;
+
+        case '\\':
+          characters.push('\\');
+          break;
+      }
+      characters.push(character);
+    }
+    characters.push("'");
+
+    return characters.join('');
+  },
```

Comment:
This functionally behaves about the same as `str::replaceAll("'", '\\')`, no?

Reply:
I think you could technically write this as
```ts
str.replaceAll("\\", "\\\\").replaceAll("'", "\\'");
```
Though that would be slower because you'd need to do it in two passes over the string.

I haven't perf tested it, but given that there are going to be hundreds of these strings - each with a few hundred characters, I think that double handling will add up.